recipes/qt/all/conanfile.py

```diff
@@ -0,0 +1,691 @@
+
+            if tools.cross_building(self.settings, skip_x64_x86=True):
+                raise ConanInvalidConfiguration("Cross compiling Qt WebEngine is not supported")
+
+            if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
+                raise ConanInvalidConfiguration("Compiling Qt WebEngine with gcc < 5 is not supported")
+
+        if self.settings.os == "Android" and self.options.opengl == "desktop":
+            raise ConanInvalidConfiguration("OpenGL desktop is not supported on Android. Consider using OpenGL es2")
+
+        if self.settings.os != "Windows" and self.options.opengl == "dynamic":
+            raise ConanInvalidConfiguration("Dynamic OpenGL is supported only on Windows.")
+
+        if self.options.with_fontconfig and not self.options.with_freetype:
+            raise ConanInvalidConfiguration("with_fontconfig cannot be enabled if with_freetype is disabled.")
+
+        if self.settings.os == "Macos":
+            del self.settings.os.version
+
+        if self.options.multiconfiguration:
+            del self.settings.build_type
+
+        if not self.options.with_doubleconversion and str(self.settings.compiler.libcxx) != "libc++":
+            raise ConanInvalidConfiguration('Qt without libc++ needs qt:with_doubleconversion. '
+                                            'Either enable qt:with_doubleconversion or switch to libc++')
+
+        if tools.os_info.is_linux:
+            if self.options.qtwebengine:
+                self.options.with_fontconfig = True
+
+        def _enablemodule(mod):
+            if mod != 'qtbase':
+                setattr(self.options, mod, True)
+            for req in self._submodules[mod]["depends"]:
+                _enablemodule(req)
+
+        for module in self._submodules:
+            if module != 'qtbase' and getattr(self.options, module):
+                _enablemodule(module)
+
+    def requirements(self):
+        if self.options.openssl:
+            self.requires("openssl/1.1.1g")
+        if self.options.with_pcre2:
+            self.requires("pcre2/10.33")
+
+        if self.options.with_glib:
+            self.requires("glib/2.64.0")
+        # if self.options.with_libiconv:
+        #     self.requires("libiconv/1.16")
+        if self.options.with_doubleconversion and not self.options.multiconfiguration:
+            self.requires("double-conversion/3.1.5")
+        if self.options.with_freetype and not self.options.multiconfiguration:
+            self.requires("freetype/2.10.1")
+        if self.options.with_fontconfig:
+            self.requires("fontconfig/2.13.91")
+        if self.options.with_icu:
+            self.requires("icu/64.2")
+        if self.options.with_harfbuzz and not self.options.multiconfiguration:
+            self.requires("harfbuzz/2.6.4")
+        if self.options.with_libjpeg and not self.options.multiconfiguration:
+            self.requires("libjpeg/9d")
```

Comment:
Is it possible to chose between `libjpeg`, `libjpeg-turbo` and `mozjpeg`?

Reply:
Probably, How do other recipes do this usually ?

Reply:
Usually, I do something like this in options:

```python
    options = {
        ...
        "with_libjpeg": [None, "libjpeg", "libjpeg-turbo", "mozjpeg"],
        ...
    }
```

Then introduce some logic to inject proper dependency.

It would be easier if conan could provide some sort of virtual package to chose between them at project level, but there is no such thing yet AFAIK.

Reply:
@uilianries @SSE4 what's the recommended way to handle this one? Should we wait for a virtual package ?

Reply:
Personally, I think that virtual packages are the best way to have global switches for different, but compatible implementations

Doing this on a per-recipe case means re-implementing this details all over again and it might lead to conflicts if this isn't done once globally

However, this concept didn't really got any attention for OpenGL... 🤷

Reply:
it is now possible to choose between libjpeg and libjpeg-turbo